lib/plugins/wallet-scoring/ciphertrace/ciphertrace.js

```diff
@@ -0,0 +1,49 @@
+const axios = require('axios')
+const _ = require('lodash/fp')
+
+const { WALLET_SCORE_THRESHOLD } = require('../../../constants')
+
+const NAME = 'CipherTrace'
+const SUPPORTED_COINS = ['BTC', 'ETH', 'BCH', 'LTC', 'BNB', 'RSK']
+
+function getClient(account) {
+  if (_.isNil(account) || !account.enabled) return null
+
+  const [ctv1, username, secretKey] = account.authorizationValue.split(':')
+  if (_.isNil(ctv1) || _.isNil(username) || _.isNil(secretKey)) {
+    throw new Error('Invalid CipherTrace configuration')
+  }
+
+  const apiVersion = ctv1.slice(-2)
+  const authHeader = {
+    "Authorization": account.authorizationValue
+  }
+  return { apiVersion, authHeader }
+}
+
+function rateWallet(account, cryptoCode, address) {
+  const client = getClient(account)
+  if (!_.includes(_.toUpper(cryptoCode), SUPPORTED_COINS) || _.isNil(client)) return Promise.resolve(null)
+
+  const { apiVersion, authHeader } = client
+  const threshold = _.isNil(account) ? WALLET_SCORE_THRESHOLD : account.scoreThreshold
```

Comment:
If account is null and ciphertrace is not configured, we can't really send a request. It will fail and reject all transactions.